zetaclient/bitcoin/bitcoin_client_test.go

```diff
@@ -26,3 +37,131 @@ func TestConfirmationThreshold(t *testing.T) {
 		require.Equal(t, int64(bigValueConfirmationCount), client.ConfirmationsThreshold(big.NewInt(1000)))
 	})
 }
+
+func TestAvgFeeRateBlock828440(t *testing.T) {
+	// load archived block 828440
+	var blockVb btcjson.GetBlockVerboseTxResult
+	err := testutils.LoadObjectFromJSONFile(&blockVb, path.Join(testutils.TestDataPath, "bitcoin_block_trimmed_828440.json"))
+	require.NoError(t, err)
+
+	// https://mempool.space/block/000000000000000000025ca01d2c1094b8fd3bacc5468cc3193ced6a14618c27
+	var blockMb testutils.MempoolBlock
+	err = testutils.LoadObjectFromJSONFile(&blockMb, path.Join(testutils.TestDataPath, "mempool.space_block_828440.json"))
+	require.NoError(t, err)
+
+	gasRate, err := CalcBlockAvgFeeRate(&blockVb, &chaincfg.MainNetParams)
+	require.NoError(t, err)
+	require.Equal(t, int64(blockMb.Extras.AvgFeeRate), gasRate)
+}
+
+func TestAvgFeeRateBlock828440Errors(t *testing.T) {
+	// load archived block 828440
+	var blockVb btcjson.GetBlockVerboseTxResult
+	err := testutils.LoadObjectFromJSONFile(&blockVb, path.Join(testutils.TestDataPath, "bitcoin_block_trimmed_828440.json"))
+	require.NoError(t, err)
+
+	t.Run("block has no transactions", func(t *testing.T) {
+		emptyVb := btcjson.GetBlockVerboseTxResult{Tx: []btcjson.TxRawResult{}}
+		_, err := CalcBlockAvgFeeRate(&emptyVb, &chaincfg.MainNetParams)
+		require.Error(t, err)
+		require.Equal(t, "block has no transactions", err.Error())
```

Comment:
We can use ErrorContains for these checks and next ones

Reply:
https://github.com/zeta-chain/node/pull/1747/commits/2124f357744661b6646c324d2cb090cb27f844c1